x-pack/plugins/log_explorer/public/components/log_explorer/log_explorer.tsx

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ScopedHistory } from '@kbn/core-application-browser';
+import { DataPublicPluginStart, ISearchStart, ISessionService } from '@kbn/data-plugin/public';
+import { DiscoverStart } from '@kbn/discover-plugin/public';
+import React, { useMemo } from 'react';
+import {
+  createLogExplorerProfileCustomizations,
+  CreateLogExplorerProfileCustomizationsDeps,
+} from '../../customizations/log_explorer_profile';
+import { createPropertyGetProxy } from '../../utils/proxies';
+
+export interface CreateLogExplorerArgs extends CreateLogExplorerProfileCustomizationsDeps {
+  discover: DiscoverStart;
+}
+
+export interface LogExplorerProps {
+  scopedHistory: ScopedHistory;
+}
+
+export const createLogExplorer = ({
+  core,
+  data,
+  discover: { DiscoverContainer },
+}: CreateLogExplorerArgs) => {
+  const customizeLogExplorer = createLogExplorerProfileCustomizations({ core, data });
+
+  return ({ scopedHistory }: LogExplorerProps) => {
+    const overrideServices = useMemo(
+      () => ({
+        data: createDataServiceProxy(data),
+      }),
+      []
+    );
```

Comment:
**Question:** there is a specific need to create this object in the scope of the returned component instead of defining it in the outer scope?
```ts
const customizeLogExplorer = createLogExplorerProfileCustomizations({ core, data });
const overrideServices = {
  data: createDataServiceProxy(data)
}

return ({ scopedHistory }: LogExplorerProps) => {
...
```